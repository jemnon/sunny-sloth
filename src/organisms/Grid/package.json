{
  "author": "Fender Digital",
  "description": "Fender Design System Grid",
  "license": "UNLICENSED",
  "main": "index.js",
  "module": "es/index.js",
  "types": "index.d.ts",
  "name": "grid",
  "scripts": {}
}