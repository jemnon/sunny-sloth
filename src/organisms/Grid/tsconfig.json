{
  "extends": "../../../tsconfig.json",
  "include": ["./"],
  "exclude": ["./__tests__", "./__mocks__", "./stories"],
  "compilerOptions": {
    "allowJs": false,
    "noEmit": false,
    "declaration": true,
    "isolatedModules": false,
    "outDir": "./lib"
  }
}