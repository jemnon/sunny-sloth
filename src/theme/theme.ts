const theme = {
  breakpoints: {
    sm: '480px',
    md: '960px',
    lg: '1280px',
    xlg: '1440px',
  },
  fonts: {
    latoBold:
      `"Lato-Bold", -apple-system, BlinkMacSystemFont, 'avenir next', avenir, 'Helvetica Neue', ` +
      `helvetica, ubuntu, roboto, noto, 'Segoe UI', Arial, sans-serif`,
    lato:
      `"Lato", -apple-system, BlinkMacSystemFont, 'avenir next', avenir, 'Helvetica Neue', ` +
      `helvetica, ubuntu, roboto, noto, 'Segoe UI', Arial, sans-serif`,
    latoItalic:
      `"Lato-Italic", -apple-system, BlinkMacSystemFont, 'avenir next', avenir, 'Helvetica Neue', ` +
      `helvetica, ubuntu, roboto, noto, 'Segoe UI', Arial, sans-serif`,
    noto:
      `"Noto Serif TC", -apple-system, BlinkMacSystemFont, 'avenir next', avenir, 'Helvetica Neue', ` +
      `helvetica, ubuntu, roboto, noto, 'Segoe UI', Arial, sans-serif`,
    notoBold:
      `"Noto Serif TC-Bold", -apple-system, BlinkMacSystemFont, 'avenir next', avenir, 'Helvetica Neue', ` +
      `helvetica, ubuntu, roboto, noto, 'Segoe UI', Arial, sans-serif`,
  },
  colors: {
    cream: '#f4ede6',
    darkTeal: '#162936',
    teal: '#244c53',
    lightBrown: '#d5a188',
    nearBlack: '#111',
    nearWhite: '#F9F9F9',
    darkOrange: '#BB5B34',
    orange: '#cf7651',
    white: '#ffffff',
    red: '#ff4136',
    green: '#19a974',
  },
  maxWidths: {
    sm: '480px',
    md: '960px',
    lg: '1280px',
  },
  spacing: {
    sm1: '0.25rem' /* 4px */,
    sm2: '0.5rem' /*  8px */,
    sm3: '0.75rem' /* 12px */,
    sm4: '1rem' /* 16px */,
    md1: '1.25rem' /* 20px */,
    md2: '1.5rem' /* 24px */,
    md3: '1.75rem' /* 28px */,
    md4: '2rem' /* 32px */,
    lg1: '2.25rem' /* 36px */,
    lg2: '2.5rem' /* 40px */,
    lg3: '2.75rem' /* 44px */,
    lg4: '3rem' /* 48px */,
  },
  transition: 'all 0.47s ease',
  zIndex: {
    z0: '0',
    z1: '1',
    z2: '2',
    z3: '3',
    z4: '4',
    z5: '5',
    z6: '6',
    z999: '999',
    z9999: '9999',
  },
};

export default theme;
