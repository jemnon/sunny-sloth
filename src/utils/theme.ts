import { DefaultTheme } from 'styled-components';

const theme: DefaultTheme = {
  breakpoints: {
    /* TODO: remove once conversion to styled-breakpoints is done */
    desktop: 'screen and (min-width: 960px)',
    tablet: 'screen and (min-width: 480px) and (max-width: 959px)',
    mobile: 'screen and (max-width: 479px)',
    /* ^^ */
    sm: '480px',
    md: '960px',
    lg: '1440px',
    xlg: '1920px',
  },
  fonts: {
    latoBold:
      `"Lato Bold", -apple-system, BlinkMacSystemFont, 'avenir next', avenir, 'Helvetica Neue', ` +
      `helvetica, ubuntu, roboto, noto, 'Segoe UI', Arial, sans-serif`,
    lato:
      `"Lato", -apple-system, BlinkMacSystemFont, 'avenir next', avenir, 'Helvetica Neue', ` +
      `helvetica, ubuntu, roboto, noto, 'Segoe UI', Arial, sans-serif`,
    latoItalic:
      `"Lato Italic", -apple-system, BlinkMacSystemFont, 'avenir next', avenir, 'Helvetica Neue', ` +
      `helvetica, ubuntu, roboto, noto, 'Segoe UI', Arial, sans-serif`,
    noto:
      `"Noto Serif TC", -apple-system, BlinkMacSystemFont, 'avenir next', avenir, 'Helvetica Neue', ` +
      `helvetica, ubuntu, roboto, noto, 'Segoe UI', Arial, sans-serif`,
  },
  colors: {
    cream: '#f4ede6',
    darkTeal: '#162936',
    teal: '#244c53',
    lightBrown: '#d5a188',
    nearBlack: '#111',
    nearWhite: '#F9F9F9',
    darkOrange: '#BB5B34',
    orange: '#cf7651',
    white: '#ffffff',
    red: '#ff4136',
    green: '#19a974',
  },
  maxWidths: {
    sm: '480px',
    md: '960px',
    lg: '1440px',
  },
  spacing: {
    s1: '0.25rem' /* 4px */,
    s2: '0.5rem' /*  8px */,
    s3: '0.75rem' /* 12px */,
    s4: '1rem' /* 16px */,
    s5: '1.25rem' /* 20px */,
    s6: '1.5rem' /* 24px */,
    s7: '1.75rem' /* 28px */,
    s8: '2rem' /* 32px */,
    s9: '2.25rem' /* 36px */,
    s10: '2.5rem' /* 40px */,
    s11: '2.75rem' /* 44px */,
    s12: '3rem' /* 48px */,
  },
  transition: 'all 0.47s ease',
  zIndex: {
    z0: '0',
    z1: '1',
    z2: '2',
    z3: '3',
    z4: '4',
    z5: '5',
    z6: '6',
    z999: '999',
    z9999: '9999',
  },
};

export default theme;
