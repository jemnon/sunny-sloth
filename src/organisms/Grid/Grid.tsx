import styled from 'styled-components';
import { Spacing } from '../../types/theme';

type Alignment =
  | 'start'
  | 'end'
  | 'center'
  | 'stretch'
  | 'space-around'
  | 'space-between'
  | 'space-evenly'
  | 'normal';
type GapSpacing = keyof Spacing;
type Flow = 'row' | 'column' | 'dense' | 'row dense' | 'column dense';

const getFR = (value: number | string): number | string =>
  typeof value === 'number' ? `repeat(${value}, 1fr)` : value;

interface GridProps {
  alignContent?: Alignment;
  areas?: string[];
  columns?: string | number;
  columnGap?: GapSpacing;
  flow?: Flow;
  gap?: GapSpacing;
  justifyContent?: Alignment;
  height?: string;
  minRowHeight?: string;
  rows?: string | number;
  rowGap?: GapSpacing;
}

const Grid = styled.section<GridProps>`
  display: grid;

  grid-auto-flow: ${({ flow = 'row' }): string => flow};
  grid-auto-rows: ${({ minRowHeight = '1rem' }): string =>
    `minmax(${minRowHeight}, auto)`};

  grid-template-rows: ${({ rows }): string =>
    rows ? `${getFR(rows)}` : 'none'};
  grid-template-columns: ${({ columns = 12 }): string => {
    return `${getFR(columns)}`;
  }};
  grid-template-areas: ${({ areas }): string =>
    areas ? areas.map(area => `"${area}"`).join(' ') : 'none'};

  gap: ${({ theme, gap = 'md4' }): string => theme.spacing[gap]};
  row-gap: ${({ theme, rowGap = 'md4' }): string => theme.spacing[rowGap]};
  column-gap: ${({ theme, gap = 'md4' }): string => theme.spacing[gap]};

  align-content: ${({ alignContent = 'normal' }): string => alignContent};
  justify-content: ${({ justifyContent = 'normal' }): string => justifyContent};

  height: ${({ height = 'auto' }): string => height};
`;

export default Grid;
