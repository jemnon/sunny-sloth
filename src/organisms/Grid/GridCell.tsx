import styled, { css, SimpleInterpolation } from 'styled-components';
import { Colors } from '../../types/theme';

type TextAlign = 'left' | 'center' | 'right';
type Color = keyof Colors;

interface GridCellProps {
  area?: string;
  bgColor?: Color;
  borderWidth?: string;
  borderColor?: Color;
  height?: number;
  left?: string | number;
  middle?: boolean;
  padding?: string;
  textAlign?: TextAlign;
  top?: string | number;
  width?: number | number[];
}

const MiddleCSS = css`
  display: inline-flex;
  flex-flow: column wrap;
  justify-content: center;
  justify-self: stretch;
`;

const GridCell = styled.div<GridCellProps>`
  grid-column-end: ${({ width = 1 }): string => `span ${width}`};
  grid-row-end: ${({ height = 1 }): string => `span ${height}`};

  ${({ area }): SimpleInterpolation => area && `grid-area: ${area}`};
  ${({ left }): SimpleInterpolation => left && `grid-column-start: ${left}`};
  ${({ top }): SimpleInterpolation => top && `grid-row-start: ${top}`};
  ${({ middle }): SimpleInterpolation => middle && `${MiddleCSS}`};

  text-align: ${({ textAlign = 'left' }): string => textAlign};

  background-color: ${({ bgColor }): SimpleInterpolation => bgColor ?? 'none'};

  border-color: ${({ borderColor = 'nearBlack', theme }): string =>
    theme.colors[borderColor]};
  border-width: ${({ borderWidth = '0' }): string => borderWidth};
  border-style: solid;

  padding: ${({ padding = '0' }): string => padding};

  height: 100%;
  min-width: 0;
`;

export default GridCell;
