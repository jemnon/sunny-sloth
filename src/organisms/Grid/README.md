# Grid

### Description
Component description

```js
import Grid from 'Grid';
```

### Props
* `propName` - property name and description 
