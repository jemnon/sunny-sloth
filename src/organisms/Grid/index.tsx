import Grid, { GridCell } from './Grid';

export { GridCell };
export default Grid;
