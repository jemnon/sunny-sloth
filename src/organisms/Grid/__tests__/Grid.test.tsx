import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import Grid from '../Grid';
import GridCell from '../GridCell';
import theme from '../../../theme';

describe('<Grid />', (): void => {
  const component = renderer.create(
    <ThemeProvider theme={theme}>
      <Grid columns={1}>
        <GridCell width={1}>1/1</GridCell>
      </Grid>
    </ThemeProvider>,
  );
  it('exists', (): void => {
    expect(Grid).toBeDefined();
    expect(GridCell).toBeDefined();
  });
  it('renders defaults correctly', (): void => {
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
