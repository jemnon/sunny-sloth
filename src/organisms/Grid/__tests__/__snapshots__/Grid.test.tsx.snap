// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Grid /> renders defaults correctly 1`] = `
<section
  className="sc-bdfBwQ jhIIfw"
>
  <div
    className="sc-gsTCUz fSGNbk"
    width={1}
  >
    1/1
  </div>
</section>
`;
